enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java

```diff
@@ -66,4 +78,78 @@ public static EndpointCluster processEndpoints(org.wso2.choreo.connect.discovery
         }
         return endpointCluster;
     }
+
+    /**
+     * Handles mock API call and prepares response object considering provided values in the request.
+     *
+     * @param requestContext request context
+     * @param responseObject response object for the mock API call
+     */
+    public static void processMockedApiCall(RequestContext requestContext, ResponseObject responseObject) {
+        responseObject.setDirectResponse(true);
+        String acceptType = "";
+        ResourceConfig resourceConfig = requestContext.getMatchedResourcePath();
+        MockedApiConfig mockedApiConfig = resourceConfig.getMockedApiConfig();
+        Map<String, String> headersMap = requestContext.getHeaders();
+        if (headersMap.containsKey(APIConstants.ACCEPT_HEADER.toLowerCase())) {
+            acceptType = headersMap.get(APIConstants.ACCEPT_HEADER.toLowerCase());
+        }
+
+        if (mockedApiConfig.getIn().equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
+            setMockApiResponse(responseObject, headersMap, mockedApiConfig, acceptType);
+        } else {
+            Map<String, String> queryParamMap = requestContext.getQueryParameters();
+            setMockApiResponse(responseObject, queryParamMap, mockedApiConfig, acceptType);
+        }
+    }
+
+    /**
+     * Prepares mock API response considering the properties map and mockedApiConfig.
+     *
+     * @param responseObject  Response object (represents response for the API call)
+     * @param propertiesMap   A map which includes values specified in headers or query parameters
+     * @param mockedApiConfig Holds the JSON values specified in the mock API implementation
+     * @param acceptType      Denotes accepted contend type as the response
+     */
+    private static void setMockApiResponse(ResponseObject responseObject, Map<String, String> propertiesMap,
+                                           MockedApiConfig mockedApiConfig, String acceptType) {
+        String requestValuePosition = mockedApiConfig.getIn();
+        String name;
+        if (requestValuePosition.equalsIgnoreCase(APIConstants.MockApiConstants.HEADER)) {
```

Comment:
Let's add a comment why this is necessary

Reply:
https://github.com/wso2/product-microgateway/blob/23a984c1ba605bf1799d68bc7a3f9d32fb8d1d98/enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/Utils.java#L119